api/api/router.go

```diff
@@ -259,3 +263,30 @@ func NewRouter(appCtx AppContext) (*mux.Router, error) {
 func recoveryHandler(next http.Handler) http.Handler {
 	return sentry.Recoverer(next)
 }
+
+var httpDuration = promauto.NewHistogramVec(prometheus.HistogramOpts{
+	Name: "merlin_http_duration_seconds",
+	Help: "Duration of HTTP requests.",
+}, []string{"name", "path", "user_agent"})
+
+// prometheusMiddleware implements mux.MiddlewareFunc.
+func prometheusMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		route := mux.CurrentRoute(r)
+
+		name := route.GetName()
+		path, _ := route.GetPathTemplate() // returns template (`/obj/{id}`) rather than actual path (`/obj/123`) which would avoid a cardinality explosion
+
+		userAgent := r.UserAgent()
+		ua := ""
+		if strings.Contains(userAgent, "merlin-sdk") || strings.Contains(userAgent, "python") {
+			ua = userAgent
+		}
+
+		timer := prometheus.NewTimer(httpDuration.WithLabelValues(name, path, ua))
```

Comment:
It's better to not label the duration metrics with user agent since it's already high in term of cardinality. Let's only label the counter metrics.